blog/2022/08-30-discover-kubernetes/index.md

```diff
@@ -0,0 +1,303 @@
+
+```shell
+> search is(kubernetes_pod) and name=~resoto
+kind=kubernetes_pod, id=7d2e76ef, name=resotocore-ndqfs, age=11d19h, cloud=k8s, account=dev, region=resoto
+kind=kubernetes_pod, id=7c7c6cfa, name=resotometrics-8vbwp, age=11d19h, cloud=k8s, account=dev, region=resoto
+kind=kubernetes_pod, id=6297b9e1, name=resotoworker-8rzjq, age=11d19h, cloud=k8s, account=dev, region=resoto
+```
+
+The result that we see here are three pods that have been installed on the `dev` cluster in the the `resoto` namespace. Please note that we are able to filter the resources for any predicate that is combined with `and` or `or` using brackets to define the precedence. We could also show all properties of all matching resources using the `dump` command.
+
+```shell
+> search is(kubernetes_pod) and name=~resoto | dump
+# result is stripped for brevity
+```
+
+## Get the service of a specific pod
+
+Kubernetes uses labels and selectors to define the relationship between a service and a pod. It is valid to expose the same pod via different services. To reveal which service exposes a pod, we need to understand the selector syntax and check all pods for matching labels. Thankfully, Resoto analyzes the relationships between resources, providing this information as edges between nodes and revealing a directed acyclic graph.
+
+To understand the concept of a graph a bit better, let me show you two examples that we enter into our Jupyter notebook:
+
+```
+search is(kubernetes_service) and name=test-service -[0:2]->
+```
+
+![Outbound Dependencies](img/test_service_outbound.svg)
+
+We are searching for a specific Kubernetes service with the name `test-service` and asking to traverse all outgoing dependencies from the service to a maximum depth of two. The result is a graph that shows the service and three pods behind this service, that all reference the same configmap.
+
+```
+search is(kubernetes_service) and name=test-service <-[0:2]-
+```
+
+![Inbound Dependencies](img/test_service_inbound.svg)
+
+We can do the same query, but this time reverse the direction of the arrow, showing us all incoming dependencies up to two levels. This time we see all resources that have a dependency on this service. Please also note that Resoto understands the underlying provider of this Kubernetes cluster, in this case, DigitalOcean. This service type is `LoadBalancer`, so a publicly available load balancer needs to be provided. A DigitalOcean load balancer is used here, backing the Kubernetes service. You can also see that the DigitalOcean load balancer is provided in the Frankfurt region in the default VPC. All of this knowledge is gathered automatically by Resoto without any additional action.
```

Comment:
I feel like you should make the big deal out of it that it actually is - that you can see the underlying provider. You may want to reference deferred edges, e.g. https://github.com/someengineering/resoto/pull/921 
No? 

Also, I thought we had a blog post on deferred edges.

Reply:
It only works for DigitalOcean atm - it is totally worth a blog post and making a big deal, once we support it also on AWS and GCP.